packages/js/src/plugins/auctionHouseModule/operations/buy.ts

```diff
@@ -0,0 +1,214 @@
+   * We only need a subset of the `Listing` model but we
+   * need enough information regarding its settings to know how
+   * to execute the sale.
+   *
+   * This includes, its asset, auction house address, seller, receipt address etc.
+   */
+  listing: Pick<
+    Listing,
+    | 'asset'
+    | 'auctionHouse'
+    | 'canceledAt'
+    | 'sellerAddress'
+    | 'tradeStateAddress'
+    | 'receiptAddress'
+  >;
+
+  /**
+   * The Auctioneer authority key.
+   * It is required when Auction House has Auctioneer enabled.
+   *
+   * @defaultValue No default value.
+   */
+  auctioneerAuthority?: Signer;
+
+  /**
+   * The token account address that's associated to the asset a bid created is for.
+   * If this or seller isn't provided in the Listing, then the bid will be public.
+   *
+   * @defaultValue No default value.
+   */
+  tokenAccount?: Option<PublicKey>;
+
+  /**
+   * The buyer's price.
+   *
+   * @defaultValue 0 SOLs or tokens.
+   */
+  price?: SolAmount | SplTokenAmount;
+
+  /**
+   * The number of tokens to bid for.
+   * For an NFT bid it must be 1 token.
+   *
+   * When a Fungible Asset is put on sale.
+   * The buyer can then create a buy order of said assets that is
+   * less than the token_size of the sell order.
+   *
+   * @defaultValue 1 token.
+   */
+  tokens?: SplTokenAmount;
+
+  /**
+   * Prints the bid receipt.
+   * The receipt holds information about the bid,
+   * So it's important to print it if you want to use the `Bid` model
+   *
+   * The receipt printing is skipped for the Auctioneer Auction House
+   * Since it currently doesn't support it.
+   *
+   * @defaultValue `true`
+   */
+  printReceipt?: boolean;
+
+  /**
+   * The address of the bookkeeper wallet responsible for the receipt.
+   *
+   * @defaultValue `metaplex.identity()`
+   */
+  bookkeeper?: Signer;
+
+  /** A set of options to configure how the transaction is sent and confirmed. */
+  confirmOptions?: ConfirmOptions;
+};
+
+/**
+ * @group Operations
+ * @category Outputs
+ */
+export type BuyOutput = {
+  /** Seller trade state account address encoding the listing order. */
+  sellerTradeState: PublicKey;
+
+  /** Buyer trade state account address encoding the bid order. */
+  buyerTradeState: PublicKey;
+
+  /** The buyer address. */
+  buyer: PublicKey;
+
+  /** The seller address. */
+  seller: PublicKey;
+
+  /** The asset's metadata address. */
+  metadata: PublicKey;
+
+  /** The address of the bookkeeper account responsible for the receipt. */
+  bookkeeper: Option<PublicKey>;
+
+  /** The PDA of the receipt account in case it was printed. */
+  receipt: Option<Pda>;
+
+  /** The sale price. */
+  price: SolAmount | SplTokenAmount;
+
+  /** The number of tokens bought. */
+  tokens: SplTokenAmount;
+
+  /** A model that keeps information about the Purchase. */
+  purchase: Purchase;
+
+  /** The blockchain response from sending and confirming the transaction. */
+  response: SendAndConfirmTransactionResponse;
+};
```

Comment:
The output should return the purchase and the bid/listing within the output object. It should not be inlined like this.